src/ctia/bulk/core.clj

```diff
@@ -196,24 +238,36 @@
 
 (s/defn patch-entities
   "patch many entities provided their type and returns errored and successed entities' ids"
-  [patches entity-type auth-identity params
-   services :- APIHandlerServices]
+  [patches
+   entity-type 
+   tempids :- TempIDs
+   auth-identity :- auth/AuthIdentity
+   params
+   services :- APIHandlerServices
+   {:keys [enveloped-result?]} :- {(s/optional-key :enveloped-result?) (s/maybe s/Bool)}]
   (when (seq patches)
     (let [get-fn #(read-entities %  entity-type auth-identity services)
           {:keys [realize-fn new-spec]} (get (all-entities) entity-type)]
-      (flows/patch-flow
-       :services services
-       :get-fn get-fn
-       :realize-fn realize-fn
-       :update-fn (update-fn entity-type auth-identity params services)
-       :long-id-fn #(with-long-id % services)
-       :entity-type entity-type
-       :identity auth-identity
-       :patch-operation :replace
-       :partial-entities patches
-       :spec new-spec
-       :make-result make-bulk-result
-       :get-success-entities (get-success-entities-fn :updated)))))
+      (cond-> (flows/patch-flow
+                :services services
+                :get-fn get-fn
+                :realize-fn realize-fn
+                :update-fn (update-fn entity-type auth-identity params services)
+                :long-id-fn #(with-long-id % services)
+                :entity-type entity-type
+                :identity auth-identity
+                :patch-operation :replace
+                :partial-entities patches
+                :tempids tempids
+                :spec new-spec
+                :make-result make-bulk-result
+                :enveloped-result? enveloped-result?
+                :get-success-entities (get-success-entities-fn :updated))
+        enveloped-result? (update :data #(mapv (fn [{:keys [error id] :as result}]
+                                                 (if error result id))
+                                               %))))))
```

Comment:
> It seems that you tried several options here :-).

I introduced `enveloped-result?` in desperation :) I agree with your suggestion.

Reply:
I removed `enveloped-result?` from the flows namespace. It's still in the bundle/bulk namespace because it needs to change the result of `gen-bulk-from-fn`. I couldn't figure out a good way to completely remove it.